shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -162,8 +163,17 @@ public boolean sendKeyEvent(KeyEvent event) {
     @Override
     public boolean performEditorAction(int actionCode) {
         // TODO(abarth): Support more actions.
-        mFlutterChannel.invokeMethod("TextInputClient.performAction",
-            Arrays.asList(mClient, "TextInputAction.done"));
+        switch (actionCode) {
+            case EditorInfo.IME_ACTION_NONE:
+                mFlutterChannel.invokeMethod("TextInputClient.performAction",
+                    Arrays.asList(mClient, "TextInputAction.none"));
```

Comment:
What do you think about `TextInputAction.newline`? Android's choice of 'none' sounds like a no-op, and Apple's choice of 'default' is pretty ambiguous.

Reply:
I had interpreted it as "What special action is attached to the enter button?"  But you make a good point: it's not asking the code to do *nothing*, so "none" isn't really right.  But will we *always* insert a newline as the default action for the enter button (I think the answer is probably "yes", but still)?  Maybe "TextInputAction.enter"?

Reply:
Totally worth doing a poll of nearby folks sitting nearby :-) `newline` definitely feels sort of... specific to the '\n' character as opposed to the action of inserting a line break (`linebreak`?). `enter` feels a bit ambiguous (and iOS keyboards call it `return`). 

I don't have a super-strong preference one way or another between `newline`, `linebreak`, `enter`, or `return`. @Hixie is brilliant at names though, maybe he's got an idea or preference?

Reply:
OK, after speaking with Ian, we determined that "newline" was the most acceptable of the options.